### PR TITLE
Implement Functionality to Retrieve the Current Fee for the Ledger

### PR DESCRIPTION
**Pull request by keefertaylor**
## High Level Overview of Change

Implement `getFee` for the  `XpringClient` which uses official rippled protocol buffers. 

This method is not hooked up. Test coverage will temporarily decrease. 

### Context of Change

rippled implemented protocol buffer support natively, which uses a new and incompatible set of protocol buffers. The implementation in rippled was completed in: https://github.com/ripple/rippled/pull/3159.

Protocol buffer upgrades are being done in piecemeal fashion for ease of review. 

*Status*

- [x] [Generate new versions of the protocol buffers](https://github.com/xpring-eng/Xpring-JS/pull/108)
- [x] [Migrate existing code to a `legacy` subdirectory and namespace](https://github.com/xpring-eng/Xpring-JS/pull/109)
- [ ] Create code that works with the new protocol buffers from the rippled team
   - [x] [Create a new `NetworkClient` with fakes](https://github.com/xpring-eng/Xpring-JS/pull/111)
   - [x] [Create a new `DefaultXpringClient` with tests](https://github.com/xpring-eng/Xpring-JS/pull/114/files)
   - [ ] Read only calls for new `DefaultXpringClient` (In progress in this PR)
   - [ ] Write calls fro `DefaultXpringClient`
- [x] [Add a boolean parameter to switch between implementations](https://github.com/xpring-eng/Xpring-JS/pull/130)

### Type of Change

- [x] New feature (non-breaking change which adds functionality)

## Before / After

Users who use the new protocol buffers will be able to check a transaction's status. 

## Test Plan

Unit tests on CI continue to pass.

### CODE REVIEW

File: src/default-xpring-client.ts [hunk 1]
```diff
@@ -1,3 +1,19 @@
+<<<<<<< HEAD
```
Comment (hbergren) [inline src/default-xpring-client.ts]: I'm gonna stop right here, because this can't be right 😄

**Comment by keefertaylor** [comment]
PTAL

File: src/default-xpring-client.ts [hunk 2]
```diff
@@ -174,6 +176,24 @@ class DefaultXpringClient implements XpringClientDecorator {
...
+    const getFeeResponse = await this.networkClient.getFee(getFeeRequest)
+    const fee = getFeeResponse.getDrops()
+    if (!fee) {
+      throw new Error(XpringClientErrorMessages.malformedResponse)
+    }
+
+    const minimumFee = fee.getMinimumFee()
+    if (!minimumFee) {
+      throw new Error(XpringClientErrorMessages.malformedResponse)
+    }
```
Comment (tedkalaw) [inline src/default-xpring-client.ts]: if you wanted to, you could do

```
class MalformedResponseError extends Error {}
```

then you can get tighter typing on the error that was thrown, and you could put the error message in there

Comment (keefertaylor) [reply src/default-xpring-client.ts]: https://app.asana.com/0/1143722888069470/1158635908525161

**Pull request merged**